### ISSUE TITLE
Example scripts missing

### ISSUE THREAD

**Issue by nanosonde** [owagner/logic4mqtt#1]
Could you please provide some example scripts?
Is there any special file suffix required so that logic4mqtt recognizes the file as a script?
How does the basic structure look like for a script.

**Comment by owagner** [comment]
Good point. My original idea was to publish my complete house automation as an example when it's done, but that's probably some time away. I'll knock up some examples.

The file suffix is used to determine the script language interpreter to be used for a given file. If you, for example, want to write Javascript code (executed with the Nashorn engine which is included in JVM 8), use the ".js" suffix.

The general idea for a script is to a) do initialization and b) register Event handlers and Timers. logic4mqtt will execute all scripts fully when it starts up, and then Event handlers and Timers get called whenever they are due. Example:

``` javascript
 /* Register an Event handler on a KNX button pair to control a Philips hue group */

// Turning it on...
Event.onUpdate("knx//Main/FirstFloor/Keypad Frontdoor Topleft",function(topic,val){
 Events.setValue("hue//groups/0",254); 
});

// ...and Turning it off
Event.onUpdate("knx//Main/FirstFloor/Keypad Frontdoor Topright",function(topic,val){
 Events.setValue("hue//groups/0",0); 
});
```

**Comment by nanosonde** [comment]
Ok, thanks.

Do all loaded scripts share a common global scope or has each script its own local scope?
E.g. global variables which are available in all scripts.

**Comment by owagner** [comment]
All scripts which use the same script engine (e.g. all Javascript scripts) share a global scope.

The $VARIABLE mechanism in Events is intended for sharing globals across different engines, and for persistant state (when using storeValue instead of setValue):

``` javascript
Events.storeValue("$GLOBALVAR","signpost");
```

sets the MQTT topic "logic/status/GLOBALVAR"

This mechanism also allows event handling based on variables.
